### PR TITLE
Parallelize guide fetching and indexing

### LINKED ISSUE #90: Parallelize git cloning
With the addition of translations, we clone many git repositories when indexing. We *might* get somewhat faster cloning by cloning those repositories in parallel. At least we can hope.

Ideally we'd use a threadpool provided by CDI/Quarkus, but I didn't find any.
So  I guess we should use `java.util.concurrent.CompletableFuture#supplyAsync(java.util.function.Supplier<U>)` , which relies on `ForkJoinPool.commonPool()`.

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #90 (and a bit more, since indexing now happens in parallel as well).

Better reviewed commit by commit, since the first commit in particular is quite disruptive.

### CODE REVIEW

File: src/main/java/io/quarkus/search/app/fetching/FetchingConfig.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.OptionalInt;
+
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+@ConfigMapping(prefix = "fetching")
+public interface FetchingConfig {
+    OptionalInt parallelism();
+
+    @WithDefault("30s")
```
Comment (marko-bekhta) [inline src/main/java/io/quarkus/search/app/fetching/FetchingConfig.java:13]: Just double checking, in the test configs you are passing it as `PT30s` and here just `30s` do both formats work?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -10,9 +10,13 @@ quarkusio.localized.cn.web-uri=https://cn.quarkus.io/
 quarkusio.localized.cn.git-uri=https://github.com/quarkusio/cn.quarkus.io.git
 quarkusio.localized.ja.web-uri=https://ja.quarkus.io/
 quarkusio.localized.ja.git-uri=https://github.com/quarkusio/ja.quarkus.io.git
-# Scheduler configuration
-# At 19:00:00 every day
+
+# Indexing configuration
+fetching.parallelism=8
+fetching.timeout=10m
```
Comment (marko-bekhta) [inline src/main/resources/application.properties:16]: just checking: I don't remember the exact time, but I remember it took a while to clone each site (just to make sure it fits into 10m)

**Review by marko-bekhta — APPROVED** [review]
All looks nice 😃 
It'll be more fun to debug it now 😄

File: src/main/java/io/quarkus/search/app/fetching/FetchingConfig.java (hunk 1 above)
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/fetching/FetchingConfig.java:13]: Yes they should both work. Though `30s` is a Quarkus-specific format that doesn't seem to work in `@QuarkusComponentTest`; I still need to report that.

File: src/main/resources/application.properties (hunk 2 above)
Comment (yrodiere) [reply src/main/resources/application.properties:16]: It seems to take less than 10m in prod, but we can certainly use a higher value if you think it's necessary.

**Comment by yrodiere** [comment]
Thanks, I'll merge then.

File: src/main/resources/application.properties (hunk 2 above)
Comment (marko-bekhta) [reply src/main/resources/application.properties:16]: If it is ok on prod then great 😃, Maybe it just was my connection and I do not plan to reclone it locally on each start so it's also fine 😄

**Pull request merged**